src/api/plugins/tdbp/warren_tdbp/api.py

```diff
@@ -75,14 +89,22 @@ async def get_cohort(
 
     Args:
         course_id (str): The course identifier on Moodle.
+        roles (LTIRole): The roles of the user.
+        user_id (str): The user identifier on Moodle.
         until (date): End date until when to compute the sliding window.
 
     Returns:
         Json: Lists of completed active actions per student.
 
     """
     logger.debug("Start computing 'cohort' indicator")
-    indicator = CohortIndicator(course_id=course_id, until=until)
+    # Instructors can see activity of all students
+    if any(role in ["instructor", "teacher", "staff"] for role in roles):
+        student_id = None
+    # Students can only see their activity
+    else:
+        student_id = user_id
```

Comment:
```suggestion
    # Students can only see their activity
    student_id = user_id
    # Instructors can see activity of all students
    if any(role in ["instructor", "teacher", "staff"] for role in roles):
        student_id = None
```

Reply:
And you can probably create a helper function doing this check, something like `is_instructor(role)` that returns a boolean.